src/nvim/api/vim.c

```diff
@@ -790,14 +790,14 @@ void nvim_echo(Array chunks, Boolean history, Dict(echo_opts) *opts, Error *err)
     verbose_enter();
   }
 
-  msg_multihl(hl_msg, history ? "echomsg" : "echo", history);
+  msg_multihl(hl_msg, opts->err ? "echoerr" : history ? "echomsg" : "echo", history, opts->err);
```

Comment:
I guess this is "echoerr" instead of "emsg" because this is a "user generated" error, not an exception, etc.

Reply:
`echoerr` to me implies that it was produced by `:echoerr` (and perhaps also by this new flag)

Reply:
> assuming that this new err flag actually plumbs through the same codepaths as :echoerr

It does. And to me the fact that non-`opts->err` invocations we're already emitted with the `echo(msg)` kind, using `echoerr` for `opts->err` seems natural.